src/Dfe.PlanTech.Application/Users/Helper/UserHelper.cs

```diff
@@ -28,9 +32,50 @@ public UserHelper(IHttpContextAccessor httpContextAccessor, IPlanTechDbContext d
             return await getUserIdQuery.GetUserId(userId);
         }
 
-        public int GetEstablishmentId()
+        public async Task<int?> GetEstablishmentId()
         {
-            return 1;
+            var getEstablishmentIdQuery = new GetEstablishmentIdQuery(_db);
+
+            var establishmentDto = _GetOrganisationData();
+
+            var reference = establishmentDto.Urn ?? establishmentDto.Ukprn;
+
+            if (reference is null)
+                throw new InvalidOperationException("Establishment reference cannot be null");
```

Comment:
Again `ArgumentNullException` might be more appropriate, and you could do something like

`new ArgumentNullException(nameof(reference))`

YMMV